In [2]:
import pandas as pd

In [51]:
df = pd.read_csv('stayzilla_sample.csv')
df

additional_info  \
0     Acceptance Rate:100 percent|Response Time:< 6 ...   
1     Acceptance Rate:100 percent|Response Time:< 6 ...   
2     Acceptance Rate:100 percent|Response Time:< 6 ...   
3     Acceptance Rate:100 percent|Response Time:< 6 ...   
4     Acceptance Rate:100 percent|Response Time:< 6 ...   
...                                                 ...   
1202  ~ All Taxes.~ Complimentary Bed Tea.~ Complime...   
1203                                        ~ All Taxes   
1204  Kindly Note ~ Hotel doesn't have Restaurants i...   
1205  ~ All taxes.~ Complimentary WiFi. And Bed Tea;...   
1206                                                NaN   

                                              amenities check_in_date  \
0                                             Newspaper    04-08-2016   
1                                            Parking|AC    04-08-2016   
2                              Pickup & Drop|Parking|AC    04-08-2016   
3           WiFi|Newspaper|Pickup & Drop|Gym|Parking|AC    04-08-2016   
4                                          Newspaper|AC    04-08-2016   
...                                                 ...           ...   
1202  Restaurent | Business Center | Wifi | Swimming...    06/01/2016   
1203                       Wifi | Laundry | Car Parking    06/01/2016   
1204                              Laundry | Car Parking    06/01/2016   
1205  Wifi | Credit Card | Laundry | Free Newspaper ...    06/01/2016   
1206          Bar/Pub | Wifi | Laundry | Free Newspaper    06/01/2016   

     check_out_date         city country  crawl_date  \
0        05-08-2016    Kalimpong   India  2016-07-20   
1        05-08-2016       Kanpur   India  2016-07-20   
2        05-08-2016      Jodhpur   India  2016-07-20   
3        05-08-2016   Jalpaiguri   India  2016-07-20   
4        05-08-2016       Kanpur   India  2016-07-20   
...             ...          ...     ...         ...   
1202     09/01/2016        Mandi     NaN         NaN   
1203     09/01/2016     Bhilwara     NaN         NaN   
1204     09/01/2016  Gandhinagar     NaN         NaN   
1205     09/01/2016   Uttarkashi     NaN         NaN   
1206     09/01/2016          Goa     NaN         NaN   

                                            description  \
0     Sangsey Valley Resort is Located at the wonder...   
1     What should you know? Enjoy unmatched service ...   
2     What should you know? A budget hotel, this acc...   
3     What should you know? The Riverwood Forest Ret...   
4     What should you know? Located at a walking dis...   
...                                                 ...   
1202                                                NaN   
1203                                                NaN   
1204                                                NaN   
1205                                                NaN   
1206                                                NaN   

                                        highlight_value  hotel_star_rating  \
0                                                   NaN                NaN   
1                                                   NaN                NaN   
2                                                   NaN                NaN   
3                                                   NaN                NaN   
4                                                   NaN                NaN   
...                                                 ...                ...   
1202  What should you know?\n Midway resort, as the ...                0.0   
1203  What should you know?\n A budget hotel that of...                0.0   
1204  What should you know?\n Hotel Shreeji boasts o...                0.0   
1205                                                NaN                0.0   
1206                                                NaN                0.0   

      ...           query_time_stamp               room_price  \
0     ...  2016-07-20 10:59:55 +0000  3167per night incl. tax   
1    

In [52]:
df['cleaned_price'] = df['room_price'].str.replace('per night incl. tax', '')
df['cleaned_price'] = pd.to_numeric(df['cleaned_price'], errors='coerce').fillna(0).astype(int)
# df.head()
# df.info()

In [53]:
print(df['cleaned_price'])

0       3167
1        815
2       1624
3       3509
4       5802
        ... 
1202       0
1203       0
1204       0
1205       0
1206       0
Name: cleaned_price, Length: 1207, dtype: int64


In [56]:
print(df['occupancy'])

0       3 Adults 2 Kids
1       2 Adults 2 Kids
2       2 Adults 2 Kids
3        1 Adult 2 Kids
4       2 Adults 2 Kids
             ...       
1202                NaN
1203                NaN
1204                NaN
1205                NaN
1206                NaN
Name: occupancy, Length: 1207, dtype: object


In [57]:
df[['adults', 'kids']] = df['occupancy'].str.extract(r'(?i)(\d+) adults?\s*(\d*) kids?')
df[['adults', 'kids']] = df[['adults', 'kids']].apply(pd.to_numeric, errors='coerce').fillna(0).astype(int)
# df.head()

In [62]:
# How does occupancy affect the room price of a property?
# Linear regression
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
import statsmodels.api as sm


In [59]:
df['cleaned_price'].info()
df['adults'].info()
df['kids'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 1207 entries, 0 to 1206
Series name: cleaned_price
Non-Null Count  Dtype
--------------  -----
1207 non-null   int64
dtypes: int64(1)
memory usage: 9.6 KB
<class 'pandas.core.series.Series'>
RangeIndex: 1207 entries, 0 to 1206
Series name: adults
Non-Null Count  Dtype
--------------  -----
1207 non-null   int64
dtypes: int64(1)
memory usage: 9.6 KB
<class 'pandas.core.series.Series'>
RangeIndex: 1207 entries, 0 to 1206
Series name: kids
Non-Null Count  Dtype
--------------  -----
1207 non-null   int64
dtypes: int64(1)
memory usage: 9.6 KB


In [63]:
X = df[['adults', 'kids']]
y = df['cleaned_price']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)

In [66]:
lr_model = sm.OLS(y, sm.add_constant(X)).fit()
print(lr_model.summary())


                            OLS Regression Results                            
Dep. Variable:          cleaned_price   R-squared:                       0.044
Model:                            OLS   Adj. R-squared:                  0.042
Method:                 Least Squares   F-statistic:                     27.65
Date:                Thu, 27 Feb 2025   Prob (F-statistic):           1.82e-12
Time:                        12:40:59   Log-Likelihood:                -11775.
No. Observations:                1207   AIC:                         2.356e+04
Df Residuals:                    1204   BIC:                         2.357e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        964.0661    319.272      3.020      0.0

Linear regression equation is: 
y = 964.07 + (673.08 * x1) + (6.33 * x2)
Adults has a p-value of 0.000 which is statistically significant since it is less than 0.05. While kids has a p-value of 0.973 which is not statistically significant. Meaning that kids has does have as strong influence on room prices. 